File: 1.two-sum.ts
/*
 * @lc app=leetcode id=1 lang=typescript
 *
 * [1] Two Sum
 */

// @lc code=start

const numIndex = new Map<number, number>();

function twoSum(nums: number[], target: number): number[] {
    let twoSumResult;
    nums.forEach((num, i) => {
        if(numIndex.has(target - num)){
            twoSumResult = [numIndex.get(target - num), i];
        }
        numIndex.set(num, i);
    })
    return twoSumResult;
};
// @lc code=end

File: 1.two-sum.go
/*
 * @lc app=leetcode id=1 lang=golang
 *
 * [1] Two Sum
 */

// @lc code=start
m := make(map[int]int)

func twoSum(nums []int, target int) []int {
	var twoSumResult
	for i, v := range nums {
		elem, ok = m[target - num]
		if ok {
			twoSumResult = [elem, i]
		}
		m[i] = v
		}
	return twoSumResult
}
// @lc code=end

